### ISSUE TITLE
Pod security policy for 1.15 concern

### ISSUE THREAD

**Issue by mo-saeed** [Azure/aks-engine#1467]
**Describe the bug**
This is regarding this https://github.com/Azure/aks-engine/blob/master/parts/k8s/addons/1.15/kubernetesmasteraddons-pod-security-policy.yaml

I am not sure why group "system:authenticated" is on both cluster-role-binding "psp:restricted" and "psp:privileged", any reason for that ?

**Comment by CecileRobertMichon** [comment]
@jackfrancis who added this in https://github.com/Azure/aks-engine/pull/1454

**Comment by mo-saeed** [comment]
also based on this https://kubernetes.io/docs/reference/access-authn-authz/rbac/#core-component-roles 
is it system:node or system:nodes ?

**Comment by aramase** [comment]
@mo-saeed This is one of the reasons I found in the aks doc for using `system: authenticated` in both privileged and restricted -

```
As shown in the following condensed output, the psp:restricted ClusterRole is assigned to any system:authenticated users. This ability provides a basic level of restrictions without your own policies being defined.
```

The policies are evaluated in alphabetical order. So the user can add their custom policies to take preference over the ones we define. For `privileged` cluster role binding, it's `system:nodes`. I did test this and the addon pods don't show up in `kube-system` unless `system:nodes` group is added to the list. This addon file that exists was used as reference for this - https://github.com/Azure/aks-engine/blob/master/parts/k8s/addons/kubernetesmasteraddons-pod-security-policy.yaml

Let me know if you think we can tune it better.
